lib/stitches/running_stitch.py

```diff
@@ -265,17 +294,46 @@ def path_to_curves(points: typing.List[Point], min_len: float):
     return curves
 
 
-def running_stitch(points, stitch_length, tolerance):
-    # Turn a continuous path into a running stitch.
+def even_running_stitch(points, stitch_length, tolerance):
+    # Turn a continuous path into a running stitch with as close to even stitch length as possible
+    # within the tolerance (including the first and last segments).
+    # This should not be used for stitching tightly-spaced parallel curves
+    # as it tends to produce ugly moiré effects in those situations.
+    # In these situations, random_running_stitch sould be used even if the maximum stitch length range is a single value.
     if not points:
         return
     stitches = [points[0]]
     for curve in path_to_curves(points, 2 * tolerance):
-        # segments longer than twice the tollerance will usually be forced by it, so set that as the minimum for corner detection
+        # segments longer than twice the tolerance will usually be forced by it, so set that as the minimum for corner detection
+        check_stop_flag()
         stitches.extend(stitch_curve_evenly(curve, stitch_length, tolerance))
     return stitches
 
 
+def random_running_stitch(points, min_stitch_length, max_stitch_length, tolerance, random_seed):
+    # Turn a continuous path into a running stitch with as close to even stitch length as possible
+    # within the tolerance (including the first and last segments).
+    # This should not be used for stitching tightly-spaced parallel curves
+    # as it tends to produce ugly moiré effects in those situations.
+    # In these situations, random_running_stitch sould be used even if the maximum stitch length range is a single value.
```

Comment:
Just adding a note here to make sure we remember to adjust this comment before marking this PR as ready for review.

Reply:
done